### PR TITLE
[MINOR][DOCS] Remove SparkSession constructor invocation in the example

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to Remove SparkSession constructor invocation in the example.

While I am here, I piggyback and add an example of Spark Connect.

### Why are the changes needed?

SparkSession's constructor is not meant to be exposed to the end users. This is also hidden in Scala side.

### Does this PR introduce _any_ user-facing change?

Yes, it removes the usage of SparkSession constructor in the user facing docs.

### How was this patch tested?

Linters should verify the changes in the CI.

### CODE REVIEW

File: python/pyspark/sql/session.py [hunk 1]
```diff
@@ -178,11 +178,6 @@ class SparkSession(SparkConversionMixin):
     ...         .config("spark.some.config.option", "some-value")
     ...         .getOrCreate()
     ... )
```
Comment (allisonwang-db) [inline python/pyspark/sql/session.py:180]: Should we add another example for creating a remote Spark session?

**Review by allisonwang-db — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/session.py [hunk 2]
```diff
@@ -179,10 +179,15 @@ class SparkSession(SparkConversionMixin):
...
 
-    Create a Spark session from a Spark context.
+    Create a Spark session with Spark Connect.
 
-    >>> sc = spark.sparkContext
-    >>> spark = SparkSession(sc)
+    >>> spark = (
+    ...     SparkSession.builder
+    ...         .remote("sc://localhost")
```
Comment (bjornjorgensen) [inline python/pyspark/sql/session.py:186]: add the port her?

**Review by LuciferYang — APPROVED** [review]
LGTM

File: python/pyspark/sql/session.py (hunk 2 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/session.py:186]: That's fine without a port - it uses the default port 15002.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.4.

**Review by dongjoon-hyun — COMMENTED** [review]
+1, LGTM.

**Pull request closed without merging**